dot/parachain/statement-distribution/statement_store.go

```diff
@@ -0,0 +1,340 @@
+package statementdistribution
+
+import (
+	"errors"
+	"slices"
+
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+)
+
+var errValidatorUnknown = errors.New("validator unknown")
+
+type statementOrigin byte
+
+func (o statementOrigin) isLocal() bool {
+	return o == statementOriginLocal
+}
+
+const (
+	// The statement originated locally.
+	statementOriginLocal statementOrigin = iota
+	// The statement originated from a remote peer.
+	statementOriginRemote
+)
+
+type storedStatement struct {
+	stmt           *parachaintypes.SignedStatement
+	knownByBacking bool
+}
+
+type fingerprintKind byte
+
+const (
+	// CompactSeconded is a fingerprint for a Seconded statement.
+	fingerprintKindCompactSeconded fingerprintKind = iota
+	// CompactValid is a fingerprint for a Valid statement.
+	fingerprintKindCompactValid
+)
+
+type fingerprint struct {
+	validator     parachaintypes.ValidatorIndex
+	kind          fingerprintKind
+	candidateHash parachaintypes.CandidateHash
+}
+
+type validatorMeta struct {
+	groupIdx       parachaintypes.GroupIndex
+	withinGroupIdx uint
+	secondedCount  uint
+}
+
+type groupStatements struct {
+	seconded parachaintypes.BitVec
+	valid    parachaintypes.BitVec
+}
+
+func newGroupStatements(size int) (*groupStatements, error) {
+	seconded, err := parachaintypes.NewBitVec(slices.Repeat([]bool{false}, size))
+	if err != nil {
+		return nil, err
+	}
+
+	valid, err := parachaintypes.NewBitVec(slices.Repeat([]bool{false}, size))
```

Comment:
same as explained above
```suggestion
	valid, err := parachaintypes.NewBitVec(make([]bool, size))
```